javascript/javascript1/week3/homework/item-arrey-removal.js

```diff
@@ -0,0 +1,28 @@
+
+function removeItemFromArray(names,nameToRemove) {
+    for( let i =0; i < names.length; i++) {
+        if (names[i] == nameToRemove) {
+            names.splice(i, 1);
+            i--; //to handle removal of duplicate elements in arrey
+        } 
```

Comment:
Instead of a for loop, Javascript can do the searching for you.
```suggestion
const index = names.indexOf(nameToRemove);
names.splice(index, 1);
```

Reply:
Thanks Cameron. Good to learn about this feature of Java Script. However, it doesn't precisely cover what my code block does, as it handles the removal of duplicate entries as well (e.g. if there were 2 "Ahmad" in the array. So a kind of a loop will be needed anyways.
Also, the for loop gives more control compared to indexOf. For example, assume that in the future we want to make the finding and removal of the elements case-insensitive.

Reply:
You are right about the two Ahmads. :)